src/renderer/components/ft-list-lazy-wrapper/ft-list-lazy-wrapper.js

```diff
@@ -36,11 +36,33 @@ export default Vue.extend({
   computed: {
     hideLiveStreams: function() {
       return this.$store.getters.getHideLiveStreams
+    },
+    hideUpcomingPremieres: function () {
+      return this.$store.getters.getHideUpcomingPremieres
     }
   },
   methods: {
     onVisibilityChanged: function (visible) {
       this.visible = visible
+    },
+
+    showResult: function (data) {
+      if (!data.type) {
+        return false
+      }
+      if (data.type === 'video') {
+        if (this.hideLiveStreams && (data.liveNow || data.lengthSeconds == null)) {
+          // hide livestreams
+          return false
+        }
+        if (this.hideUpcomingPremieres &&
+            (data.isUpcoming || data.premiere || data.durationText === 'PREMIERE' ||
+             (data.isRSS && data.viewCount === '0'))) {
```

Comment:
Not sure if we should include the rss check as there could be videos that aren't premiers that will be hidden

Reply:
I guess this is `(data.isRSS && data.viewCount === '0')`
Meaning if using RSS feed and view count is zero, **assume** it's premiere

Reply:
Pika is correct. This is the only feasible way to remove them in the case of RSS data--no other flags to pivot on.

Reply:
I've added some additional comments alongside this to clarify.